pyproject.toml

```diff
@@ -21,7 +21,7 @@ classifiers = [
 repository = "https://github.com/numaproj/numaflow-python"
 
 [tool.poetry.dependencies]
-python = ">=3.9, <3.13"
+python = ">=3.9, <4.0"
```

Comment:
Make it < 3.14.
Things might break otherwise in future

Reply:
+1 Python changes a lot between the minor versions :)

Reply:
TBH, it is very difficult to support 5 versions of Python simultaneously and have trouble free builds. I would vote for dropping support for 3.9 if we are adding 3.13.
Anyway 3.9 is going obsolete after Oct 2025.

Reply:
+1 yank support for 3.9

Reply:
> Make it < 3.14. 
> Things might break otherwise in future

I'm against limiting the version of Python.
Please read the following: https://iscinumpy.dev/post/bound-version-constraints/
If you still think there's a good reason to block 3.14+, please share it.

Note that supported versions will still be listed in classifiers... so if somebody uses a version not listed, then they use at their own risk.

Reply:
I'm ambivalent about dropping support for Python 3.9, so I'll refrain from giving an opinion on that.